File: reverse-array/ReverseArray.cs
using System;
using System.Collections.Generic;

public class ReverseArray{
    public static int[] ReverseAnArray1(int[] array) {
        int[] result = new int[array.Length];
        for(int i = array.Length - 1; i >= 0; i--) {
           result[i] = array[array.Length - i - 1];
        }
        return result;
    }

    public static int[] ReverseAnArray2(int[] array) {
        for(int i = 0; i < array.Length / 2; i++) {
            int lastIndex = array.Length - i - 1;
            int temp = array[lastIndex];
            array[lastIndex] = array[i];
            array[i] = temp;
        }
        return array;
    }

    public static int[] ReverseAnArray3(int[] array) {
        int left = 0;
        int right = array.Length - 1;
        while(left < right) {
            int temp = array[left];
            array[left] = array[right];
            array[right] = temp;
            left++; right--;
        }
        return array;
    }
}

File: reverse-array/ReverseArray.java
import java.util.*;

public class ReverseArray {
    public static int[] reverseArray1(int[] array) {
        int[] result = new int[array.length];
        for(int i = array.length - 1; i >= 0; i--) {
            result[i] = array[array.length - i - 1];
        }
        return result;
    }

    public static int[] reverseArray2(int[] array) {
        for(int i = 0; i < array.length / 2; i++) {
            int lastIndex = array.length - i - 1;
            int temp = array[i];
            array[i] = array[lastIndex];
            array[lastIndex] = temp;
        }
        return array;
    }

    public static int[] reverseArray3(int[] array) {
        int left = 0;
        int right = array.length - 1;
        while(left < right) {
            int temp = array[left];
            array[left] = array[right];
            array[right] = temp;
            left++; right--; 
        }
        return array;
    }
}